src/ripple/RippleEngine.ts

```diff
@@ -275,6 +303,155 @@ export class XrpEngine extends CurrencyEngine<
     }
   }
 
+  /**
+   * Parse TakerGets or TakerPays into an EdgeAssetAmount
+   * */
+  parseRippleDexTxAmount = (
```

Comment:
Replace most of parseRippleDexTxAmount and processRippleDexTx with getBalances to determine src/dest amounts of order fulfillments.

Reply:
The way I had it done is more accurate and complete according to [xrpscan.com](http://xrpscan.com/). 

The balances object is good for including mainnet fees in the xrp amount total, but for some reason underreports the actual amounts involved in the trades (regardless of fees - token amounts are underreported). Also there's no way to tell unfilled order amounts from the `Balances` obj either.

Reply:
Comparing against xrpscan.com shows alignment between my amounts and the summary amounts given by api queries to xrp explorers